src/Symfony/Component/Console/Helper/ProgressBar.php

```diff
@@ -287,10 +285,21 @@ public function setProgress(int $step)
             $step = 0;
         }
 
-        $prevPeriod = (int) ($this->step / $this->redrawFreq);
-        $currPeriod = (int) ($step / $this->redrawFreq);
+        $prevStep = $this->step;
         $this->step = $step;
         $this->percent = $this->max ? (float) $this->step / $this->max : 0;
+
+        if (null === $this->redrawFreq) {
+            if (microtime(true) - $this->lastWriteTime >= .05 || $this->max === $step) {
```

Comment:
Correct, this changes default behaviour.

Correct, it's not possible to revert back to time-based redraw once progressbar object has been constructed and setRedrawFrequency called.

I don't think this change will cause flooding the output of any existing app,  because that happens only if app produces lot of output. If it produces lot of output and it's not flooding the output, that means app developers already use setRedrawFrequency call, which will mean new default behaviour won't apply.

Reply:
>  it's not flooding the output, that means app developers already use setRedrawFrequency call

Why so? the default right now is 1/10, so no call to setRedrawFrequency is required to prevent flooding. Did I miss something?

Reply:
1/10 default does not prevent flooding at all. If there is `2*10^6` redraws per second, reducing it to `2*10^5` does not help much. It was really arbitrarily set frequency which helps in minimum cases. Additionally, new default will help even in such cases where 1/10 is correct, since 50ms is correct as well, in all cases.

I'm trying to think about use case with frequency where redrawing every 10th time is more correct than redrawing every 50ms, but I can't find any :thinking: Can you think about some?

Reply:
the frequency is not in time but in steps: 1/10 means there will always be 10 steps whatever the $max.

Reply:
It's neither. 1/10 means redraw will happen every $max/10 time. So it triggers 10 redraws in total _only_ if $max is 100. If $max is 1000 and job takes 10 seconds to finish, there are 100 redraw calls each second with current default. It's really pretty arbitrary number and not very useful default. It might help only if by coincidence current `time of the job run/$max/10` approaches maximum writing speed of output (terminal).

Here's minimal reproducible so you can test it:
```php
<?php
require 'vendor/autoload.php';

use Symfony\Component\Console\Helper\ProgressBar;
use Symfony\Component\Console\Output\ConsoleOutput;

$progressBar = new ProgressBar(new ConsoleOutput(), $maxSteps = 100000000);
for ($i=0; $i<= $maxSteps; $i++) {
    $progressBar->advance();
}
```

Reply:
Actually you are right. However, default you are talking about is applied only when output is not decorated. I think limit to 10 redraws as default makes sense there. I've put it back.

For non-decorated output, default is that it's being redrawn on each step and I would like to change that to something more reasonable (this PR).

Reply:
Humans are used to 24, 30 and 60 frames per second ... which means a frame duration of 41ms, 33ms and 16ms. So, maybe we can change `50ms` (which seems arbitrary) to `41ms`? (Maybe 50ms is not arbitrary and there's a reason behind it).

Reply:
Redraw every 50ms corresponds to 20FPS. This is best FPS that can be achieved without hurting performance too much. Even this is too much IMO and it has negative performance impact. Remember, this is CLI application, not a movie. Even GIFs have 15FPS and they are perfectly ok - and that is for entertainment value. ProgressBar is just an indicator. If anything, frequency should be decreased. I have increased to 50ms it after complaints, but trying to go for movie like FPS in CLI is not realistic.